File: base/src/main/java/ru/badr/base/util/json/DateLongSerializer.java
package ru.badr.base.util.json;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonPrimitive;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;
import java.util.Date;

/**
 * Created by Badr on 16.11.2015.
 */
public class DateLongSerializer implements JsonDeserializer<Date>, JsonSerializer<Date> {
    @Override
    public Date deserialize(JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
        return new Date(jsonElement.getAsLong());
    }

    @Override
    public JsonElement serialize(Date date, Type type, JsonSerializationContext jsonSerializationContext) {
        JsonElement element = new JsonPrimitive(date != null ? date.getTime() : 0L);
        return element;
    }
}

File: remote/src/main/java/com/histler/base/remote/serializer/DateLongSerializer.kt
package com.histler.base.remote.serializer

import com.google.gson.*
import java.lang.reflect.Type
import java.util.*

/**
 * Created by abadretdinov
 * on 08.08.2018
 */
class DateLongSerializer : JsonDeserializer<Date>, JsonSerializer<Date> {
    @Throws(JsonParseException::class)
    override fun deserialize(jsonElement: JsonElement, type: Type, jsonDeserializationContext: JsonDeserializationContext): Date {
        return Date(jsonElement.asLong)
    }

    override fun serialize(date: Date?, type: Type, jsonSerializationContext: JsonSerializationContext): JsonElement {
        return JsonPrimitive(date?.time ?: 0L)
    }
}